crates/router/src/core/payments/helpers.rs

```diff
@@ -1434,6 +1434,18 @@ pub(crate) fn validate_payment_method_fields_present(
                 ) | (
                     api_enums::PaymentMethod::Crypto,
                     api::PaymentMethodData::Crypto(..)
+                ) | (
+                    api_enums::PaymentMethod::BankTransfer,
+                    api::PaymentMethodData::BankTransfer(..)
+                ) | (
+                    api_enums::PaymentMethod::Reward,
+                    api::PaymentMethodData::Reward(..)
+                ) | (
+                    api_enums::PaymentMethod::Upi,
+                    api::PaymentMethodData::Upi(..)
+                ) | (
+                    api_enums::PaymentMethod::Voucher,
```

Comment:
How are we going to ensure that a new payment method would be added here too?

Reply:
Right. I'll add a `foreign_from` implementation to ensure this.